core/src/main/java/org/infinispan/interceptors/locking/AbstractLockingInterceptor.java

```diff
@@ -96,14 +95,15 @@ protected Object visitNonTxDataWriteCommand(InvocationContext ctx, DataWriteComm
    @Override
    public final Object visitInvalidateCommand(InvocationContext ctx, InvalidateCommand command) throws Throwable {
       try {
-         boolean skipLocking = hasSkipLocking(command);
-         long lockTimeout = getLockAcquisitionTimeout(command, skipLocking);
-         for (Object key : command.getKeys()) {
-            lockKey(ctx, key, lockTimeout, skipLocking);
+         if (hasSkipLocking(command)) {
+            return invokeNextInterceptor(ctx, command);
          }
+         lockAllAndRecord(ctx, Arrays.stream(command.getKeys()), getLockTimeoutMillis(command));
```

Comment:
Actually I forgot ``Arrays.asList(..)`` just uses the backing array.  In that case it isn't bad - the old one created multiple collections.  But to be honest there isn't much difference between ``Arrays.asList(..).toStream()`` vs ``Arrays.asStream(..)`` except the latter is a bit more condense.

Also I think the reason @pruivo separated filtering and the locking is because not all uses require filtering, which is simple to do with streams but less efficient (collections) or not possible without multiple iterations (regular for loop).  Granted we could have a different method for each or pass in a always true filter for a single version, but I personally like the separation of ideas.  But like I said in my other comments this is all personal preference so I am fine either way :)